### PR TITLE
travis:  upload binaries

### PR DESCRIPTION
**Pull request by ken2812221**
Print out sha256 hash and upload the binaries to transfer.sh

This is the easiest way to test and verify what PR do.

1. Download it from travis-ci console
2. Verify the sha256 hash
3. Start testing the binaries

### CODE REVIEW

**Comment by scravy** [comment]
This will upload the binaries for every travis job that runs. Maybe limit it to pull requests on the bitcoin repository?

**Comment by practicalswift** [comment]
@ken2812221 Can the binaries be downloaded from Travis currently? Please update the PR description with scenarios that could be achieved post merge that cannot be achieved pre-merge :-)

**Comment by ken2812221** [comment]
> This will upload the binaries for every travis job that runs. Maybe limit it to pull requests on the bitcoin repository?

People may want to test the master branch.
> Can the binaries be downloaded from Travis CI console currently?

Yes, it prints out the URL at the end of each build.
> Please update the PR description with scenarios that could be achieved post merge that cannot be achieved pre merge :-)

What do you mean?

**Comment by practicalswift** [comment]
@ken2812221 Oh, sorry I think I misunderstood you. The binaries are not available from download directly from Travis CI, but by uploading them to `transfer.sh` they are made available? If so it makes sense. I think the text "Download it from travis-ci console" fooled me into thinking the binaries were downloadable directly from Travis CI :-)

File: .travis.yml [hunk 1]
```diff
@@ -130,6 +130,7 @@ script:
     - if [ "$TRAVIS_EVENT_TYPE" = "cron" ]; then extended="--extended --exclude feature_pruning,feature_dbcrash"; fi
     - if [ "$RUN_TESTS" = "true" ]; then BEGIN_FOLD functional-tests; DOCKER_EXEC test/functional/test_runner.py --combinedlogslen=4000 --coverage --quiet --failfast ${extended}; END_FOLD; fi
 after_script:
+    - test ! -z $OUTDIR && for file in $(find $OUTDIR/bin -executable -type f); do curl --upload-file $file https://transfer.sh/$(sha256sum $file | cut -d' ' -f1)-$(basename $file); echo; done
```
Comment (practicalswift) [inline .travis.yml]: Please add flags to `curl` so that we don't fail in case of unavailability of `transfer.sh`. Please handle both timeout and downtime.

Comment (ken2812221) [reply .travis.yml]: Done

**Comment by scravy** [comment]
> People may want to test the master branch.

@ken2812221 fair enough, but it will also run for every fork (other peopls repositories), every branch, every push that anyone does, right?

**Comment by scravy** [comment]
I think the curl version in the travis image is quite an old one (the one shipped with ubuntu 14.04 trusty probably).

I see in the logs:

```
0.38s$ test ! -z $OUTDIR && for file in $(find $OUTDIR/bin -executable -type f); do curl --retry 3 --retry-delay 0 -m 60 --connect-timeout=10 --upload-file $file https://transfer.sh/$(sha256sum $file | cut -d' ' -f1)-$(basename $file); echo; done
curl: option --connect-timeout=10: is unknown
curl: try 'curl --help' or 'curl --manual' for more information
curl: option --connect-timeout=10: is unknown
curl: try 'curl --help' or 'curl --manual' for more information
curl: option --connect-timeout=10: is unknown
curl: try 'curl --help' or 'curl --manual' for more information
curl: option --connect-timeout=10: is unknown
curl: try 'curl --help' or 'curl --manual' for more information
curl: option --connect-timeout=10: is unknown
curl: try 'curl --help' or 'curl --manual' for more information
```

But I think `after_script` sections don't fail the build (?) hence all checks passed.

**Comment by ken2812221** [comment]
@scravy I should not use `=`. That's my mistake.

**Comment by ken2812221** [comment]
> but it will also run for every fork (other peopls repositories), every branch, every push that anyone does, right?

I can't see any downside of doing this.

**Comment by MarcoFalke** [comment]
Hmm, we could really limit this to pull requests on our repo. Otherwise, the hoster might exclude travis ips for spamming, no?

**Comment by ken2812221** [comment]
>we could really limit this to pull requests on our repo. Otherwise, the hoster might exclude travis ips for spamming.

You really convinced me. I don't want to get any ban by any host.

**Comment by promag** [comment]
> This is the easiest way to test and verify what PR do.

I feel that the correct way to do that is to checkout and build the PR branch. It's common to tweak the code while testing, specially the corresponding tests.

At the moment, from the reviewer/tester point of view, I don't see advantages of this and for that reason NACK.

**Comment by ken2812221** [comment]
@promag It's easy to do testing and tweak the code for Linux, but it's complicate for Windows and cross compile things. The motivatation is that no body want to cross compile, I think this PR is good for that.

**Comment by laanwj** [comment]
This has been discussed before, and I don't think it's a good idea. Uploading binaries for pull requests automatically gives e.g. malware authors a way to make semi-genuine-looking executables easily.

Furthermore, the Travis executables aren't really intended to be distributed. They might, at least theoretically, be specially adapted for testing in a specific environment. It's not a gitian build.

I think @jonasschnelli / @marcofalke 's solution is better - have a real gitian build be triggered by a tag or otherwise, by maintainers.

**Comment by ken2812221** [comment]
@laanwj Thanks for your comment. Although this PR was rejected, but I can still provide my own gitian-build result in my PR, right?

**Comment by laanwj** [comment]
> but I can still provide my own gitian-build result in my PR, right?

Sure, you can do what you want in your own PRs. My argument is only against doing it automatically for PRs in a "centralized" way.

**Pull request closed without merging**